CreeDictionary/API/search.py

```diff
@@ -433,13 +432,14 @@ def fetch_lemma_by_user_query(user_query: str, **extra_constraints) -> CreeAndEn
     cree_results: Set[CreeResult] = set()
 
     # there will be too many matches for some shorter queries
-    if len(user_query) > settings.AFFIX_SEARCH_THRESHOLD:
+    if len(user_query) > settings.AFFIX_SEARCH_THRESHOLD and affix_search:
         # prefix and suffix search
-        ids_by_prefix = Wordform.affix_searcher.search_by_prefix(user_query)
-        ids_by_suffix = Wordform.affix_searcher.search_by_suffix(user_query)
+        ids_by_prefix = list(Wordform.affix_searcher.search_by_prefix(user_query))
+        ids_by_suffix = list(Wordform.affix_searcher.search_by_suffix(user_query))
 
+        # todo: this needs refactoring, our affix searcher now also return entries matched by English
         for wf in Wordform.objects.filter(
-            id__in=set(chain(ids_by_prefix, ids_by_suffix)), **extra_constraints
+                id__in=set(ids_by_prefix + ids_by_suffix), **extra_constraints
         ):
             cree_results.add(CreeResult(wf.analysis, wf, wf.lemma))
```

Comment:
I realized this while adding optional affix search. Initially I don't think this causes any difference. But I guess it does produces undesired ranking of the entries. Because EnglishResults and CreeResults are ranked differently. Here EnglishResults matched by affix search are wronly marked as CreeResults.

Reply:
Change from chain to list is accidentaly though - I did this for better debugging experience while I investigated a bug.